### ISSUE TITLE
Animation ignores selector

### ISSUE THREAD

**Issue by munzili** [AvaloniaUI/Avalonia#5488]
The following Code should create two rectangles where one is flashing yellow. Somehow both rectangles are flashing. If i change the Fill color of BlackField to a named color instead of a HEX value, the animations is working as expected. 

Following XAML reproduce the problem (create empty MVVM project and copy XAML to main window):
```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        d:DesignWidth="200" d:DesignHeight="50"
        x:Class="AvaloniaApplication.Views.MainWindow">
  <Window.Styles>
    <Style Selector="Rectangle.BlackField">
      <Setter Property="Fill" Value="#FF333333" />
    </Style>
    <Style Selector="Rectangle.Flashing">
      <Style.Animations>
        <Animation Duration="0:0:1" PlaybackDirection="Alternate">          
          <KeyFrame Cue="100%">
            <Setter Property="Fill" Value="Yellow"/>
          </KeyFrame>
        </Animation>
      </Style.Animations>
    </Style>
  </Window.Styles>
  <Canvas VerticalAlignment="Top" HorizontalAlignment="Left" Width="100" Height="50" Canvas.Left="0" Canvas.Top="0">
    <Rectangle Classes="BlackField Flashing" Canvas.Top="0" Canvas.Left="0" Width="50" Height="50"/>
    <Rectangle Classes="BlackField" Canvas.Top="0" Canvas.Left="150" Width="50" Height="50"/>
  </Canvas> 
</Window>
```

Expected behavior is only the rectangle with class "Flashing" should animate, not both!

- OS: Windows 10, VS 2019 
- Version 0.10, .net5

**Comment by jmacato** [comment]
You should add the BlackField class together with the Flashing class in the selector to make it work:
`Selector="Rectangle.BlackField.Flashing"`

**Comment by munzili** [comment]
> You should add the BlackField class together with the Flashing class in the selector to make it work:
> `Selector="Rectangle.BlackField.Flashing"`

Changing the selector to Selector="Rectangle.BlackField.Flashing" doesnt change anything on the result. Same wired behaviour. It shouldnt make also a difference as CSS-Style selector work like this - combineable classes/styles, If i create another Style: 

```xml
    <Style Selector="Rectangle.RedField">
      <Setter Property="Fill" Value="#FF333333" />
    </Style>
```

and i want to add another rectangle with classes "RedField Flashing" it should also work as excpected. Only Rectangle with classes "Flashing" should flash/blink.

**Comment by bundeskanzler114** [comment]
Same problem here (https://github.com/AvaloniaUI/Avalonia/issues/5478).

**Comment by grokys** [comment]
The main issue fixed on master by #6032. There remains a problem with the animation system not picking up the correct neutral value meaning that an additional keyframe is needed:

```
                    <KeyFrame Cue="0%">
                        <Setter Property="Fill" Value="#FF333333"/>
                    </KeyFrame>
                    <KeyFrame Cue="100%">
                        <Setter Property="Fill" Value="Yellow"/>
                    </KeyFrame>
```

But that's a separate problem, I'll close this issue and we'll track the neutral value in #6063.
